javascript/javascript2/week3/script.js

```diff
@@ -0,0 +1,96 @@
+/*window.addEventListener("DOMContentLoaded", () => {
+    // DOM ready! Images, frames, and other subresources are still downloading.
+    
+});*/
+//1
+
+function checkLoad(){
+    setTimeout(() => {
+        console.log("Called after 2.5 seconds");
+    
+    }, 2500);  
+}
+checkLoad();
+//2
+function logOutMessages(delay,stringToLogms) {
+    setTimeout(() => {
+        console.log(stringToLogms);
+    }, delay*1000); 
+}
+logOutMessages(5,"this string logged after 5 seconds");
+logOutMessages(3,"this string logged after 3 seconds");
+//3
+const logButton=document.getElementById("log-button");
+logButton.addEventListener("click",()=>{logOutMessages(5,"button ->called after 5 seconds");});
+
+//4
+const earthLogger = function(){
+  console.log("Earth");
+}
+
+const saturnLogger = function(){
+    console.log("Saturn");
+}
+function planetLogFunction(planet){
+    planet();
+}
+planetLogFunction(earthLogger);
+planetLogFunction(saturnLogger);
+
+/*function planetLogFunction(planet){
+    planet;
+}
+planetLogFunction(earthLogger());
+planetLogFunction(saturnLogger());
+*/
+
+//5
+
+const getLocationButton = document.getElementById("getLocation");
+getLocationButton.addEventListener("click",()=>{
+    if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(showPosition,error);
+  } else { 
+    console.log("Geolocation is not supported by this browser.");
+  }});
```

Comment:
Search online for `visual studio code format document` to format your code consistently.